modules/rest_api/controllers/services/rest.php

```diff
@@ -1627,12 +1960,29 @@ private function esGetSpecialFieldDataCleanerIcons(array $doc) {
    *   Elasticsearch document.
    *
    * @return string
-   *   Formatted value including website and survey dataset info.
+   *   Formatted value including website, survey dataset and,
+   *   optionally, group info.
    */
-  private function esGetSpecialFieldDatasourceCode(array $doc) {
+  private function esGetSpecialFieldDatasourceCode(array $doc, $params) {
+    if (count($params) > 1) {
+      return 'Incorrect params for non-standard datasource field (must be 0 or 1)';
+    }
     $w = $doc['metadata']['website'];
     $s = $doc['metadata']['survey'];
-    return "$w[id] ($w[title]) | $s[id] ($s[title])";
+    if (count($params) && $params[0] === 'with_group') {
```

Comment:
I'm being picky I know (!) but it feels slightly odd that adding the format specifier with_group also removes the ids from the returned string. I would have thought there needs to be a separate control over whether the ids are included in the substrings that are combined to make the output.

Reply:
Okay I will revisit.

Reply:
I've changed this so that the special field handler accepts a format parameter which is a pattern for the output. For the backward compatibility, the pattern is '<wt> | <st> {|} <gt>' and for the default, the pattern is '<wi> (<wt>) | <si> (<st>)'. So the hander replaces the <tokens> with the appropriate website, dataset or group attribute. The curly braces surround anything which should be removed if it occurs at the end of the value after substitutions. (I'll document it properly in docs - along with other necessary changes - when this is all sorted.)